rostwitter/python/rostwitter/twitter.py

```diff
@@ -54,24 +58,80 @@ def _request_url(self, url, verb, data=None):
             )
         return 0  # if not a POST or GET request
 
-    def post_update(self, status):
-        if len(status) > 140:
-            rospy.logwarn('tweet is too longer > 140 characters')
-            status = status[:140]
-        url = 'https://api.twitter.com/1.1/statuses/update.json'
-        data = {'status': StringIO(status)}
-        json = self._request_url(url, 'POST', data=data)
-        data = simplejson.loads(json.content)
+    def _check_post_request(self, request):
+        valid = True
+        data = simplejson.loads(request.content)
+        if request.status_code != 200:
+            rospy.logwarn('post tweet failed. status_code: {}'
+                          .format(request.status_code))
+            if 'errors' in data:
+                for error in data['errors']:
+                    rospy.logwarn('Tweet error code: {}, message: {}'
+                                  .format(error['code'], error['message']))
+            valid = False
+        if valid:
+            return data
+
+    def _post_update_with_reply(self, texts, media_list=None,
+                                in_reply_to_status_id=None):
+        split_media_list = []
+        media_list = media_list or []
+        for i in range(0, int(math.ceil(len(media_list) / 4.0))):
+            split_media_list.append(media_list[i * 4:(i + 1) * 4])
+        for text, media_list in zip_longest(texts, split_media_list):
+            text = text or ''
+            media_list = media_list or []
+            url = 'https://api.twitter.com/1.1/statuses/update.json'
+            data = {'status': text}
+            media_ids = self._upload_media(media_list)
+            if len(media_ids) > 0:
+                data['media_ids'] = media_ids
+            if in_reply_to_status_id is not None:
+                data['in_reply_to_status_id'] = in_reply_to_status_id
+            r = self._request_url(url, 'POST', data=data)
+            data = self._check_post_request(r)
+            if data is not None:
+                in_reply_to_status_id = data['id']
+        return data
+
+    def _upload_media(self, media_list):
+        url = 'https://upload.twitter.com/1.1/media/upload.json'
+        media_ids = []
+        for media in media_list:
+            data = {'media': media}
+            r = self._request_url(url, 'POST', data=data)
```

Comment:
in order to post `numpy array` data, we need to upgrade `requests` python module to `2.19.0` and above.
https://github.com/psf/requests/commit/8546a15587b2a240c1d1404d3ef99365b911bf13